test/conversion_test.go

```diff
@@ -0,0 +1,879 @@
+    command: ['/bin/bash']
+    args: ['-c', 'gcloud auth activate-service-account --key-file /var/secret/bucket-secret/bucket-secret-key']
+    workingDir: /dir
+    env:
+    - name: MY_VAR1
+      value: foo
+    resources:
+      inputs:
+      - name: workspace
+        resource: source-repo
+      outputs:
+      - name: workspace
+        resource: source-repo
+    volumeMounts:
+      - name: messages
+        mountPath: /messages
+    imagePullPolicy: IfNotPresent
+    securityContext: 
+      runAsUser: 2000
+    timeout: 60s
+    secret:
+      secretName: test-ssh-credentials
+    onError: continue
+  stepTemplate:
+    image: gcr.io/google.com/cloudsdktool/cloud-sdk:alpine
+    command: ['/bin/bash']
+    env:
+    - name: QUX
+      value: original
+    args: ['-c', 'gcloud auth activate-service-account --key-file /var/secret/bucket-secret/bucket-secret-key']
+    workingDir: /dir
+    env:
+    - name: MY_VAR1
+      value: foo
+    resources:
+      inputs:
+      - name: workspace
+        resource: source-repo
+      outputs:
+      - name: workspace
+        resource: source-repo
+    volumeMounts:
+      - name: messages
+        mountPath: /messages
+    imagePullPolicy: IfNotPresent
+    securityContext: 
+        runAsUser: 2000
+  sidecars:
+  - name: server
+    image: alpine/git:v2.26.2
+    command: ['/bin/bash']
+    args: ['-c', 'gcloud auth activate-service-account --key-file /var/secret/bucket-secret/bucket-secret-key']
+    workingDir: /dir
+    env:
+    - name: MY_VAR1
+      value: foo
+    resources:
+      inputs:
+      - name: workspace
+        resource: source-repo
+      outputs:
+      - name: workspace
+        resource: source-repo
+    readinessProbe:
+      periodSeconds: 1
+    volumeMounts:
+    - name: messages
+      mountPath: /messages
+    securityContext:
+      runAsUser: 0
+    script: echo test
+  volumes:
+    - name: messages
+      emptyDir: {}
+  params:
+  - name: PARAM
+    description: param des
+    type: string
+    default: "1"
+  workspaces:
+  - name: workspace
+    description: description
+    mountPath: /foo
+    readOnly: true
+    optional: true
+`, taskName, namespace))
+
+	if _, err := c.V1beta1TaskClient.Create(ctx, v1beta1Task, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create Task: %s", err)
+	}
+
+	v1beta1Task, err := c.V1beta1TaskClient.Get(ctx, taskName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected Task for %s: %s", taskName, err)
+	}
+
+	if d := cmp.Diff(v1beta1Task, v1beta1TaskExpected, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-got, +want: %v", d)
+	}
+
+	v1TaskGot, err := c.V1TaskClient.Get(ctx, taskName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Couldn't get expected Task for %s: %s", taskName, err)
+	}
+
+	if d := cmp.Diff(v1TaskGot, v1Task, filterTypeMeta, filterObjectMeta); d != "" {
+		t.Fatalf("-got, +want: %v", d)
+	}
+}
+
+func TestTaskRunCRDConversion(t *testing.T) {
```

Comment:
It's better to add comments here (as well as other two tests) to clarify that this test is testing the conversion from v1beta1 to v1, similar to what you did for `TestTaskCRDConversion`.